crates/configuration/src/shared/node.rs

```diff
@@ -95,7 +104,14 @@ impl Serialize for NodeConfig {
             state.serialize_field("command", &self.command)?;
         }
 
-        if self.args.is_empty() || self.args == self.chain_context.default_args {
+        if self.args.is_empty()
+            || self.args()
+                == self
+                    .chain_context
+                    .default_args
+                    .iter()
+                    .collect::<Vec<&Arg>>()
+        {
```

Comment:
Why modify this code ? We don't need to use getter because we have access to self fields and PartialEq borrow automatically here: https://doc.rust-lang.org/nightly/std/cmp/trait.PartialEq.html

```suggestion
        if self.args.is_empty() || self.args == self.chain_context.default_args {
```